asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py

```diff
@@ -0,0 +1,311 @@
+- Number of posts per year;
+- Distribution of posts per user;
+- Distribution of replies per user;
+- Distribution of interactions per user.
+
+Computed statistics are logged and appear in your terminal.
+Plots are saved in the folder `asf_public_discourse_home_decarbonisation/outputs/figures/mse/`.
+
+To run this script:
+`python asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py`
+
+To change category or date/time range, use the following arguments:
+`python asf_public_discourse_home_decarbonisation/analysis/mse/eda_mse_category_data.py --category <category> --collection_date_time <collection_date_time>`
+where
+<category>: category or sub-forum to be analysed, e.g. "energy", "lpg-heating-oil-solid-other-fuels", etc.
+<collection_date_time>: data collection date/time in the format YYYY_MM_DD
+"""
+
+# ## Package imports
+import warnings
+
+# Ignoring warnings from Pandas
+warnings.simplefilter(action="ignore")
+
+import pandas as pd
+from datetime import datetime
+import matplotlib.pyplot as plt
+import altair as alt
+import os
+import asf_public_discourse_home_decarbonisation.config.plotting_configs as pc
+from asf_public_discourse_home_decarbonisation.getters.mse_getters import get_mse_data
+from asf_public_discourse_home_decarbonisation import PROJECT_DIR
+import argparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+def add_datetime_variables(mse_data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Transform the datetime variable into a datetime object and add year and year_month variables.
+    Args:
+        mse_data (pd.DataFrame): dataframe with MSE data
+    Returns:
+        pd.DataFrame: Pandas DataFrame with datetime variables added
+    """
+    mse_data["datetime"] = mse_data["datetime"].apply(
+        lambda x: datetime.strptime(x, "%Y-%m-%dT%H:%M:%S+00:00")
+    )
+    mse_data["year"] = mse_data["datetime"].apply(lambda x: x.year)
+    mse_data["year_month"] = mse_data["datetime"].dt.to_period("M")
+
+    return mse_data
+
+
+def create_statistics_from_data(mse_data: pd.DataFrame):
+    """
+    Create and prints statistics from the MSE data.
+
+    Args:
+        mse_data (pd.DataFrame): _description_
+    """
+    if len(mse_data) == len(mse_data.drop_duplicates()):
+        logger.info("There are no duplicated posts in the data.")
+    else:
+        logger.warning(
+            f"There are duplicated posts in the data.\nNumber instances: {len(mse_data)}\nNumber unique instances: {len(mse_data.drop_duplicates())}"
+        )
+
+    logger.info(f"Categories in data: {mse_data['category'].unique()}")
+
+    logger.info(
+        f"Number of instances (posts & replies) collected for category '{category}': {len(mse_data)}"
+    )
+
+    number_original_posts = len(mse_data[mse_data["is_original_post"] == 1])
+    logger.info(
+        f"Number of original posts collected for category '{category}': {number_original_posts}"
+    )
```

Comment:
Is this code only ever run for one category at a time?

Reply:
yeah, you can run it for each category separately or for all data at once.

Reply:
At some stage we can create a shell script to just execute over all categories at once :)

Reply:
So do these info logging statements work with multiple categories? It looks like they would only work for one category at a time?

Reply:
@crispy-wonton - the way this code is organised is so that you can run on it on each individual category or on all categories data at once. If all categories at once, then category will have the value "all". I can add some more documentation on that to make that more obvious.